includes/customers/customer-actions.php

```diff
@@ -11,6 +11,117 @@
  * @since       3.0
  */
 
+/**
+ * Is intended to be used when adding a customer direclty with edd_add_customer function.
+ *
+ * Because the edd_add_customer function only directly interacts with the customers DB table, we may
+ * need to do some additional actions like adding items to the customer email addresses table.
+ *
+ * @since 3.0.4
+ */
+function edd_process_customer_added( $customer_id, $data ) {
+	// Make sure we add a new primary email address to the email addrsses table.
+	edd_add_customer_email_address(
+		array(
+			'customer_id' => $customer_id,
+			'email'       => $data['email'],
+			'type'        => 'primary',
+		)
+	);
+}
+add_action( 'edd_customer_added', 'edd_process_customer_added', 10, 2 );
+
+/**
+ * Is intended to be used when updating a customer direclty with edd_add_customer function.
+ *
+ * Because the edd_update_customer function only directly interacts with the customers DB table, we may
+ * need to do some additional actions like managing email addresses.
+ *
+ * @since 3.0.4
+ */
+function edd_process_customer_updated( $customer_id, $data, $previous_customer_data ) {
+	$customer = edd_get_customer( $customer_id );
+
+	// If the email address changed, set the new one as primary.
+	if ( $previous_customer_data->email !== $customer->email ) {
+
+		// Our transition hook for the type will handle demoting any other email addresses.
+		edd_add_customer_email_address(
+			array(
+				'customer_id' => $customer->id,
+				'email'       => $customer->email,
+				'type'        => 'primary',
+			)
+		);
+	}
+	// Process a User ID change.
+	if ( intval( $customer->user_id ) !== intval( $previous_customer_data->user_id ) ) {
+		// Attach the User Email to the customer as well.
+		$user = new WP_User( $customer->user_id );
+		if ( $user instanceof WP_User ) {
+			edd_add_customer_email_address(
+				array(
+					'customer_id' => $customer->id,
+					'email'       => $user->user_email,
+					'primary'     => true, // Set this as the primary, which should transition any other emails.
+				)
+			);
+		}
+
+		// Remove the old user email from this account.
+		$previous_user = new WP_User( $previous_customer_data->user_id );
+		if ( $previous_user instanceof WP_User ) {
+			edd_delete_customer_email_address(
+				array(
+					'customer_id' => $customer->id,
+					'email'       => $previous_user->user_email,
+				)
+			);
+		}
+
+		// Update some payment meta if we need to.
+		$order_ids = edd_get_orders( array( 'customer_id' => $customer->id, 'number' => 9999999 ) );
+
+		foreach ( $order_ids as $order_id ) {
+			edd_update_order( $order_id, array( 'user_id' => $customer->user_id ) );
```

Comment:
Are you intending for the orders to still have the previous_user email address assigned? If not, this should also update the order email address.

Reply:
@robincornett I'm not sure exactly honestly. We previously did not alter the email address on the order. I think since we use that email address for resending the purchase receipt as an option in the radio items, we should probably leave it on the order.